streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java

```diff
@@ -402,6 +404,9 @@ private void captureOutputRecords() {
         // Capture all the records sent to the producer ...
         final List<ProducerRecord<byte[], byte[]>> output = producer.history();
         producer.clear();
+        if (streamProcessingGuarantee != null && streamProcessingGuarantee.equals(StreamsConfig.EXACTLY_ONCE)) {
```

Comment:
We do not need the first check since `streamsConfig.getString(StreamsConfig.PROCESSING_GUARANTEE_CONFIG)` will always return a non-null string to `streamProcessingGuarantee`

Reply:
fixed